### ISSUE TITLE
"Documentation"

### ISSUE THREAD

**Issue by MarquisDeGris** [TensorBox/TensorBox#257]
Hi,

I've downloaded the framework and started first tests.

Thanks for the effort!!

My current main concern is that I'm not sure if I can actually use this framework for my task at hand. Mainly because there is no documentation (that I can see). 

Can I train the the framework to find arbitrary objects (some that are not in the pretrained database), e.g., the break lever on a train.
If so how many pictures do have to provide to train for my object.
After a learning phase of, e.g., 300K iterations can I "ask" the program where the lever is for a new picture? How do I do this precisely?
There are Test/Train/Val jsons. I would guess train is the one that holds the rectangles for the images that I provide to "teach" the program what to look for.
Test would hold the images that I want to have predictions for, but why would I have to provide this json? And that does val do exactly?

As you can see there is a lot of confusion at this point. And I'm sure some of these questions have been asked/answered already. I would gladly start a FAQ section as soon as I actually understand what I am talking about :)

So please, someone, enlighten me. 

Kind Regards

**Comment by MarquisDeGris** [comment]
Found something
that's a start
https://github.com/TensorBox/TensorBox/issues/188

**Comment by RomanSteinberg** [comment]
Hi, I'll try to answer your questions. But first of all I want to point that TensorBox is not a framework at all. Tensorflow is a framework with lots of documentation. Tensorbox is project to train some object detection models.
1. Documentation. The only documentation the author provided is the `readme` file. 
2. Prediction. You can do it, and if read the `readme` you would know it.
3. Test/Train/Val sets. You have to take some of the Deep Learning courses. The description of this sets division is about correct model `construction`. TensorBox is like any other model should satisfy those requirements.

If you ask me, I would not allow questions like your in the FAQ section. People who wish to use projects like TensorBox should learn DL, ML,  frameworks (Tensorflow) and language (C++, Python) used. These are a prerequisites.

**Comment by RomanSteinberg** [comment]
@MarquisDeGris do you satisfied? Can we close the issue?

**Comment by MarquisDeGris** [comment]
Nope, I ain't. But you won't help as you made clear. "Taking classes about deep learning" does NOT answer questions about your program. 
By the way, what is a framework in your opinion and why would you consider TensorBox not to be a framework to find arbitrary objects?

**Comment by AdamWP** [comment]
Yes, you can train to detect a train lever.  You just need a lot of images of train levers.  I trained a custom model using 5,000 images with 20,000 objects that I wanted to train for. I let it train for 1,000,000.  I think this was overkill though.    I recommend you experiment.  Try 1000 images at 100K iterations.  You will see in tensorboard if your training is converging and improving.    train.py trains the model, evaluate.py evaluates your model against your evaluation dataset showing results.    Read the evaluation part of the readme, it give you the specific training and evaluation command to run.
